### ISSUE TITLE
Administrator group should be able to modify all content

### ISSUE THREAD

**Issue by mjgiarlo** [samvera/hyrax#167]
<a href="https://github.com/vantuyls"><img src="https://avatars.githubusercontent.com/u/9534576?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [vantuyls](https://github.com/vantuyls)**
_Thursday Sep 01, 2016 at 22:10 GMT_
_Originally opened as https://github.com/projecthydra/sufia/issues/2614_

----

Administrator can: 
- [ ] search and view all items (including private items not owned by me), 
- [ ] edit all items (including items I do not own), 
- [ ] delete all items, 
- [ ] transfer ownership of all items (including items I do not own), 
- [ ] share all items (including items I do not own), 
- [ ] move items to and from all collections and admin sets

Decomposes #2421

**Comment by mjgiarlo** [comment]
<a href="https://github.com/mjgiarlo"><img src="https://avatars.githubusercontent.com/u/131982?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mjgiarlo](https://github.com/mjgiarlo)**
_Friday Oct 28, 2016 at 23:09 GMT_

----

Out of scope for current mediated deposit sprints.

**Comment by mjgiarlo** [comment]
@hannahfrost @vantuyls @scholarworks How high a priority is this issue?

**Comment by hannahfrost** [comment]
@mjgiarlo I think it is high priority.

**Comment by jcoyne** [comment]
Has anyone done analysis on this to determine which of these already work and which do not?

**Comment by jcoyne** [comment]
What does this mean? 
> move items to and from all communities and sets

**Comment by jcoyne** [comment]
What does "ownership" mean?  Hyrax doesn't model ownership, only people with edit access.

**Comment by vantuyls** [comment]
@jcoyne i fixed that "communities and sets" stuff. but, basically, if an items is in the "wrong" admin set or collection, an admin can move them.

As for "ownership", i'm referring to who has the object as "My Work" - see attachment:

![screen shot 2017-02-02 at 12 22 42 pm](https://cloud.githubusercontent.com/assets/9534576/22566851/7114ae62-e942-11e6-8a02-62ccbf8e023f.png)

**Comment by jcoyne** [comment]
@vantuyls "My Work" is currently filtering by depositor: https://github.com/projecthydra-labs/hyrax/blob/1e504c200fd9c39120f514ac33cd42cd843de9fa/app/search_builders/hyrax/search_builder.rb#L8-L13
 
So we can either add an "ownership" concept to the model, or change who deposited an item.  Either of these seem like major changes and should probably be handled as a new ticket.

**Comment by jcoyne** [comment]
@vantuyls do we have any sense that "search & view", "edit" and "delete" all items, doesn't already work?

**Comment by jcoyne** [comment]
As "sharing ownership" in it's present implementation would be impossible because an object can only have a single depositor.

**Comment by mjgiarlo** [comment]
I assume that meant "share file/work" in the edit view, rather than "share ownership" given the ill defined notion of ownership.

**Comment by mjgiarlo** [comment]
@jcoyne Just did some testing. here's where things stand:

- [ ] search and view items I lack explicit access to, and that may be private, when NOT in workflow
- [ ] search and view items in workflow, when I do not explicitly participate in that workflow
- [ ] edit items to which I was not explicitly granted access
- [ ] delete items to which I was not explicitly granted access
- [ ] transfer ownership of items to which I was not explicitly granted access
- [ ] share items to which I was not explicitly granted access
- [ ] move items (to which I was not explicitly granted access) to and from all collections and admin sets

**Comment by mjgiarlo** [comment]
So, in summary, administrative users can do almost everything with items they don't have explicit access to, except:

* Download the file
* Search/view/edit the work IF SUPPRESSED
* Transfer ownership, because the work does not appear in the transfer UI which lives in the dashboard

**Comment by mjgiarlo** [comment]
@hannahfrost how much of the above should be in the MVP? We can decompose this ticket into more granular ones.

**Comment by hannahfrost** [comment]
@mjgiarlo 

- What does "SUPPRESSED" mean? That seems like a problem to be fixed for MVP.
- Download the file is MVP
- Transfer ownership problem may be solved already. Presumably the Admin could search for the work to be transferred using the search field on All Works panel (see https://github.com/projecthydra-labs/hyku/issues/447). Your thoughts, @ggeisler?

**Comment by mjgiarlo** [comment]
@hannahfrost By suppressed I mean that the work is currently in a workflow state that is not published. So in other words, it's not Hydra/Fedora ACLs that are preventing administrative users from seeing these works; it's our (search builder) code that handles who sees works that are currently "in workflow." (I put that in quotes because technically all works are always in workflow.)

Re: transfers, there is currently only one spot in the UI to initiate a transfer and that's in "My Works." Administrators do not see works to which they lack explicit access in their "My Works" view. The All Works design by @ggeisler would remedy this, you're right!

So it sounds like we need two new tickets for MVP.

**Comment by mjgiarlo** [comment]
@hannahfrost Apologies if I'm just spacing here... does the third bullet above also mean we should discuss how much of https://github.com/projecthydra-labs/hyku/issues/447 is included in MVP? Or are we comfy with punting transfers (and other bits of hyku 447) off 'til post-MVP?

**Comment by hannahfrost** [comment]
@mjgiarlo Thanks for definition of suppressed.

With respect to the Works design, I think we should discuss what of Hyku#447 is in scope for MVP. I feel like we had an exchange to that effect, but not sure where now.

**Comment by ggeisler** [comment]
We had a brief exchange on Slack the other day to the effect of Hyku#447 should be decomposed into multiple development tickets, only some of which need need be prioritized for MVP, so we need to figure out which ones are MVP. Nothing more than that, unless @hannahfrost and @mjgiarlo had additional discussions in another venue after that.

**Comment by mjgiarlo** [comment]
I thought it would be good to verify the behavior in this issue, last summarized as follows: 
> So, in summary, administrative users can do almost everything with items they don't have explicit access to, except:
> 
> * Download the file
> * Search/view/edit the work IF SUPPRESSED
> * Transfer ownership, because the work does not appear in the transfer UI which lives in the dashboard

While verifying this behavior, I found the following:

1. As an admin user, when a work is deposited by another user into the default Admin Set without mediation:
  1. The title of uploaded files on the work show page shows up as `File`
  1. The file actions dropdown for each file on the work show page is empty (no download, edit, or delete)
1. As an admin user, when a work is deposited by another user into the default Admin Set **with mediation**:
  1. The work show page renders as unavailable (because suppressed)
  1. The work does not appear in the `Under Review` tab in the `Review Submissions` page
  1. (Both of the above issues are resolved by adding the `approving` role to the admin user in the `Workflow Roles` view, which should **not** be required)

Thus, I put in the following PRs
* https://github.com/samvera/hyrax/pull/1327, which restores the download button for admin users
* https://github.com/samvera/hyrax/pull/1328, which restores the edit and delete buttons for admin users, and the FileSet title issue
* https://github.com/samvera/hyrax/pull/1331, which fixes foo and bar above.

This leaves two bits of potential work described above TBD:

* As an admin user, I should be able to search for and find suppressed items from the main navigation search bar. @jcoyne, are there reasons we should not pursue this?
* Design transfers of ownership so that they're admin-aware. This needs design, and I don't believe this is a high priority at the moment. Can I get @hannahfrost and @ggeisler to :+1: that? Also, we should write this up as a ticket if we haven't yet.

Long story short: this is mostly done, and arguably done enough for now.

**Comment by mjgiarlo** [comment]
@vantuyls An update on this work: :point_up:

**Comment by ggeisler** [comment]
@mjgiarlo I'm happy to look at the TBD regarding transfers of ownership if and when it's given priority. I have no issue with it not being a design priority now.

**Comment by vantuyls** [comment]
thanks @mjgiarlo. i see that i was absent throughout Feb, sorry about that!

**Comment by hannahfrost** [comment]
@mjgiarlo @ggeisler I created https://github.com/samvera/hyrax/issues/1382. Agreed it is not a current high priority. From the admin dashboard point of view, is the design needed here similar to that needed for embargo/leases in https://github.com/samvera/hyrax/issues/1127?

**Comment by mjgiarlo** [comment]
@hannahfrost Yes, I believe so. Thanks!

**Comment by no-reply** [comment]
@vantuyls can I get your estimation of the status of this work?

**Comment by vantuyls** [comment]
Current Status: As an administrator (on Nurax) i see the following:
- Admin cannot download a Private item
- Known title search for a Private item does not return the item
- Transfer of ownership by administrator of Private items now works. 

If we still intend for Admin to have access and control over all content, the first two in that list need to be resolved.

**Comment by no-reply** [comment]
@jlhardes i think we should slate this for upcoming work, if either of these items is still outstanding.

**Comment by rjkati** [comment]
As an admin user on nurax-dev:

- Admin can download a Private item from another user
- Admin search for a known title of a Private item and that item is returned in the results

@jlhardes

**Comment by jlhardes** [comment]
Closing this issue since it looks like the reported issues for admin users have all been addressed now.
